### DISCUSSION TITLE
What does waitForPrevious do to a sourceConnector?

### DISCUSSION THREAD

**Discussion by zifr** [nextgenhealthcare/connect#5620] [Q&A] [answered]
On Mirth Connect Server 3.12.0, when I export a channel from Mirth I get this xml element.

![image](https://user-images.githubusercontent.com/7194235/213309520-a5ecc23e-3db4-474c-a970-5a285a4c5d8f.png)

I searched the codebase and the only relevant part I see is in [ChannelSetup.java](https://github.com/nextgenhealthcare/connect/blob/e0db5497aabb94c41071e4a098580e660a1d5d8e/client/src/com/mirth/connect/client/ui/ChannelSetup.java#L2662) which appears to be something else entirely: UI instructions for setting up Destinations which also have a waitForPrevious property.

My question is, does waitForPrevious affect a channel's sourceConnector in some way?

**Comment by siddharthsharma9258** [comment]
I think you are wrong. It's an option on the destination side only and not on the source side.

That option allows you work your destination one after the other. So let's say you are consuming a set of APIs begining with the Authentication end point, then you can have D1 getting the auth and D2 waiting on sucess of D1 to work.

**Reply by pacmano1** [reply]
What @siddharthsharma9258 said - but I'd modify that at bit.   D1 does not have to be successful, just complete.  So you must code for that fact (i.e. that D1 had an error and you do not want D2 to execute.)

**Reply by zifr** [reply]
Thank you for responding. I understand the waitForPrevious option and how it changes the behavior of a Destination chain. 

My concern is when I export a channel, the sourceConnector seems to also have an option waitForPrevious. I can change its value then import the channel back into Mirth, and while this doesn't seem to have any noticable effect on the channel I wanted to be sure I wasn't missing something since I see no trace of such an option in the docs or references to it in the code.

**Comment by jonbartels** [comment]
https://docs.nextgen.com/bundle/Mirth_User_Guide_41/page/connect/connect/topics/t_The_Message_Processing_Lifecycle_connect_ug.html

**Comment by tonygermano** [comment, marked as answer]
For whatever reason, waitForPrevious is a property of Connector, which is inherited by both source and destination connectors. 
https://github.com/nextgenhealthcare/connect/blob/78b20b075e1ab20d0357612006bf014b33be2e25/server/src/com/mirth/connect/model/Connector.java#L50

It only seems to be important in this loop, which only checks the value on destination connectors
https://github.com/nextgenhealthcare/connect/blob/78b20b075e1ab20d0357612006bf014b33be2e25/server/src/com/mirth/connect/server/controllers/DonkeyEngineController.java#L1198-L1202

**Reply by zifr** [reply]
I appreciate your help with confirming this.
